File: aula-08-06-2024/primeiro-app/app/src/main/java/com/example/meu_primeiro_app/MainActivity.java
package com.example.meu_primeiro_app;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import java.text.DecimalFormat;

public class MainActivity extends AppCompatActivity implements View.OnClickListener, View.OnLongClickListener {

    private TextView txtImc;
    private Button btnCalcularImc;
    private Button btnLimpar;
    private EditText edtPeso;
    private EditText edtAltura;

    // o método onCreate é o primeiro método invocado no ciclo de vida
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // a classe R faz referência a pasta res
        // a propriedade layout faz referência a minha pasta layout
        // activity_main representa o layout da minha activity
        setContentView(R.layout.activity_main);
        // mapeando os elementos de interface
        this.txtImc = findViewById(R.id.txt_imc);
        this.btnCalcularImc = findViewById(R.id.btn_calcular_imc);
        this.btnLimpar = findViewById(R.id.btn_limpar);
        this.edtPeso = findViewById(R.id.edt_peso);
        this.edtAltura = findViewById(R.id.edt_altura);
        // mapeando eventos
        this.btnCalcularImc.setOnClickListener(this);
        this.btnLimpar.setOnClickListener(this);
        // mapeando o evento de clique longo
        this.btnCalcularImc.setOnLongClickListener(this);
    }

    private void limpar() {
        this.edtPeso.setText("");
        this.edtAltura.setText("");
        this.txtImc.setText("0.0");
    }

    private void calcularImc() {
        Log.i("teste", "Caiu no método para calcular o imc!");
        double imc = 0;
        String pesoString = this.edtPeso.getText().toString().trim();
        String alturaString = this.edtAltura.getText().toString().trim();
        boolean ok = true;

        if (pesoString.isEmpty()) {
            ok = false;
            // Toast.makeText(getApplicationContext(), "Informe o peso", Toast.LENGTH_LONG).show();
            /**
             * o método setError() adicionar um efeito de erro no campo
             * com uma mensagem que você passar como parâmetro para o método
             */
            edtPeso.setError("O campo de peso deve ser preenchido!");
        }

        if (alturaString.isEmpty()) {
            ok = false;
            // Toast.makeText(getApplicationContext(), "Informe aaltura", Toast.LENGTH_LONG).show();
            edtAltura.setError("O campo altura deve ser preenchido!");
        }

        if (ok) {
            double peso = Double.parseDouble(pesoString);
            double altura = Double.parseDouble(alturaString);
            imc = peso / (altura * altura);
            Log.i("imc_calculado", String.valueOf(imc));
            DecimalFormat decimalFormat = new DecimalFormat("0.00");
            this.txtImc.setText("IMC: " + decimalFormat.format(imc));
        }

    }

    @Override
    public void onClick(View view) {
        // método para controlar o evento de clique

        if (view.getId() == this.btnCalcularImc.getId()) {
            // clicou no botão para calcular o imc
            this.calcularImc();
        } else {
            // clicou no botão para limpar os campos e o texto com o calculo do imc
            this.limpar();
        }

    }

    @Override
    public boolean onLongClick(View view) {
        // método para controlar o evento de clique longo
        Toast.makeText(getApplicationContext(), "Clique longo no botão!", Toast.LENGTH_LONG)
                .show();

        return false;
    }

}

File: aula-08-06-2024/calculadora-imc-kotlin/app/src/main/java/com/example/calculadora_imc_kotlin/MainActivity.kt
package com.example.calculadora_imc_kotlin

import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.View.OnClickListener
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity(), OnClickListener {

    private var btnLimpar: Button? = null
    private var btnCalcularImc: Button? = null
    private var txtImc: TextView? = null
    private var edtPeso: EditText? = null
    private var edtAltura: EditText? = null
    private val calculadoraImc: CalculadoraIMC = CalculadoraIMC()

    /**
     * o método onCreate() está sobreescrebendo o método
     * onCreate() da classe AppCompactActivity
     */
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        this.mapearElementosInterface()
        this.mapearEventos()
    }

    private fun mapearElementosInterface() {
        this.txtImc = findViewById(R.id.txt_imc)
        this.btnLimpar = findViewById(R.id.btn_limpar)
        this.btnCalcularImc = findViewById(R.id.btn_calcular_imc)
        this.edtPeso = findViewById(R.id.edt_peso)
        this.edtAltura = findViewById(R.id.edt_altura)
    }

    private fun mapearEventos() {
        this.btnCalcularImc!!.setOnClickListener(this)
        this.btnLimpar!!.setOnClickListener(this)
    }

    private fun limpar() {
        // o método clear vai limpar os campos
        this.edtPeso!!.text.clear()
        this.edtAltura!!.text.clear()
        this.txtImc!!.text = "0.0"
    }

    private fun validarFormulario(): Boolean {
        var ok: Boolean = true

        if (this.edtPeso!!.text.toString().trim() == "") {
            ok = false
            this.edtPeso!!.error = "Informe o peso!"
        } else if (this.edtPeso!!.text.toString().toDouble() <= 0.0) {
            ok = false
            this.edtPeso!!.error = "Peso inválido!"
        }

        if (this.edtAltura!!.text.toString().trim() == "") {
            ok = false
            this.edtAltura!!.error = "Informe a altura!"
        } else if (this.edtAltura!!.text.toString().toDouble() <= 0) {
            ok = false
            this.edtAltura!!.error = "Altura inválida!"
        }

        return ok
    }

    private fun calcularImc() {
        Log.i("teste", "Caiu no método para calcular o imc!")
        try {

            if (this.validarFormulario()) {
                // realizar o calculo do imc
                val peso: Double = this.edtPeso!!.text.trim().toString().toDouble()
                val altura: Double = this.edtAltura!!.text.trim().toString().toDouble()
                val imc: Double = this.calculadoraImc.calcular(peso, altura)
                this.txtImc!!.text = imc.toString()
            } else {
                Log.e("erro_calcular_imc", "Ocorreu um erro ao tentar-se calcular o imc!")
                Toast.makeText(applicationContext, "Erros de validação de dados!", Toast.LENGTH_LONG).show()
            }

        } catch (e: Exception) {
            Log.e("erro", e.message.toString() + " - " + e.stackTraceToString())
        }

    }

    override fun onClick(p0: View?) {

        if (p0!!.id == this.btnCalcularImc!!.id) {
            Log.d("teste", "O usuário clicou no botão para calcular o imc!")
            // o usuário clicou no botão para calcular o imc
            this.calcularImc()
        } else {
            // o usuário clicou no botão para limpar
            this.limpar()
        }

    }

}